File: БР1.1/Захматов Юрий/labs/lab1/src/main/java/com/petproject/itmoacsbackend/booking/entities/BookingEntity.java
package com.petproject.itmoacsbackend.booking.entities;

import com.petproject.itmoacsbackend.payments.entities.PaymentEntity;
import com.petproject.itmoacsbackend.property.entities.PropertyEntity;
import com.petproject.itmoacsbackend.booking.enums.BookingStatus;
import com.petproject.itmoacsbackend.users.entities.UserEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Table(name = "booking")
@Entity
@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class BookingEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "renter_id", nullable = false)
    private UserEntity renterId;

    @ManyToOne
    @JoinColumn(name = "property_id", nullable = false)
    private PropertyEntity propertyId;

    @OneToOne
    @JoinColumn(name = "payment_id")
    private PaymentEntity paymentId;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BookingStatus bookingStatus = BookingStatus.PENDING;

    @Column(nullable = false)
    private Double totalPrice;

    private Integer guestsCount;

    private String details;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }
}

File: БР1.1/Майстренко Анастасия Николаевна/labs/lab1/rental-api/src/models/booking.entity.ts
import {
    Entity,
    Column,
    PrimaryGeneratedColumn,
    ManyToOne,
    OneToMany,
    JoinColumn,
    CreateDateColumn,
    UpdateDateColumn,
    BaseEntity,
} from 'typeorm';

import { BookingStatus } from './enums';
import { Property } from './property.entity';
import { User } from './user.entity';
import { Review } from './review.entity';

@Entity()
export class Booking extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(() => Property, (property) => property.bookings, {
        nullable: false,
    })
    @JoinColumn({ name: 'propertyId' })
    property: Property;

    @Column({ type: 'int' })
    propertyId: number;

    @ManyToOne(() => User, (user) => user.bookings, { nullable: false })
    @JoinColumn({ name: 'tenantId' })
    tenant: User;

    @Column({ type: 'int' })
    tenantId: number;

    @Column({ type: 'date' })
    startDate: string;

    @Column({ type: 'date' })
    endDate: string;

    @Column({ type: 'float', default: 0 })
    totalPrice: number;

    @Column({ type: 'varchar', length: 20, default: BookingStatus.PENDING })
    status: BookingStatus;

    @OneToMany(() => Review, (review) => review.booking)
    reviews: Review[];

    @CreateDateColumn()
    createdAt: Date;

    @UpdateDateColumn()
    updatedAt: Date;
}
